### DISCUSSION TITLE
Glances cannot start in web server mode

### DISCUSSION THREAD

**Discussion by liblaf** [Homebrew/discussions#3683] [Linux]
I installed `glances` with brew on Linux. It works well in standalone mode. However, it seems not to support web server mode.

```terminal
❯ glances -w
Bottle module not found. Glances cannot start in web server mode.
```

**Comment by alebcay** [comment]
For `glances` and other Python-based packages, we automatically parse the package's required dependencies to be bundled together with the package.

In this case, the `bottle` module is needed for web server mode and is an optional dependency, thus it was excluded. The proper change to include this dependency is:
- Add the `bottle` module as a resource to [the formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/glances.rb). This should enable the functionality.
- Test that the web server mode is able to start as part of the formula's `test` block.
- Mark `bottle` as an extra dependency to be kept for `glances` in [the corresponding config file](https://github.com/Homebrew/homebrew-core/blob/master/pypi_formula_mappings.json), otherwise the next automated version update of the package will strip `bottle` away again since it's not a required dependency.

I'll get around to trying to get these in if I have a chance, but these changes are also reasonably sized for anyone who wants to try picking it up ahead of me.
